compiler/semtypinst.nim

```diff
@@ -433,6 +433,18 @@ proc handleGenericInvocation(cl: var TReplTypeVars, t: PType): PType =
   # One step is enough, because the recursive nature of
   # handleGenericInvocation will handle the alias-to-alias-to-alias case
   if newbody.isGenericAlias: newbody = newbody.skipGenericAlias
+
+  let origSym = newbody.sym
+  if origSym != nil and sfFromGeneric notin origSym.flags:
+    # same as `replaceTypeVarsS` but directly set the type without recursion:
+    newbody.sym = copySym(origSym, cl.c.idgen)
+    incl(newbody.sym.flags, sfFromGeneric)
+    newbody.sym.owner = origSym.owner
+    newbody.sym.typ = newbody
+    # unfortunately calling `replaceTypeVarsN` causes recursion, so this AST
+    # is the original generic body AST
+    newbody.sym.ast = origSym.ast
```

Comment:
Not saying it's a good idea but you could use `copyTree(origSym.ast)` here at least.

Reply:
Shouldn't matter but ok